### ISSUE TITLE
Store the queue defaultJobOptions separate from the jobs to reduce memory consumption

### ISSUE THREAD

**Issue by lukas-becker0** [taskforcesh/bullmq#2625]
I noticed that the `defaultJobOptions` will be added to each job in redis, 
I thought they would be stored in the queue separate from the jobs.

I think it would be nice to store the `defaultJobOptions` in the queue and not add them to each job in redis,
that would reduce the memory consumption when e.g. using custom job retain options for 
completed and failed jobs.

**Comment by lukas-becker0** [comment]
Okay, I missed that we can at least do this for the retain options by passing them to the worker options as opposed to the queue options although that difference has not been documented.

https://docs.bullmq.io/guide/queues/auto-removal-of-jobs
https://docs.bullmq.io/guide/workers/auto-removal-of-jobs

**Comment by manast** [comment]
Yes, we are trying to move some settings to the workers to reduce memory usage.
